src/Pages/Login/Login.js

```diff
@@ -1,25 +1,65 @@
-import React, { Component } from 'react';
+import React, { Component, useState } from 'react';
+import { Link } from 'react-router-dom';
 import styled from 'styled-components';
 
-const Login = () => {
+const Login = ({ history }) => {
+  const { Kakao } = window;
+
+  const KakaoLogin = () => {
+    Kakao.Auth.login({
+      success: function (authObj) {
+        console.log(authObj);
+        fetch('http://3.34.199.216:8000/users/kakaosignin', {
+          method: 'POST',
+          body: JSON.stringify({
+            access_token: authObj.access_token,
+            refresh_token: authObj.refresh_token,
+          }),
+        })
+          .then(res => res.json())
+          .then(res => {
+            console.log(res);
+            sessionStorage.setItem('Watta_token', res.access_token);
+            if (res.message === 'success') {
+              alert(`${res.user}님 Wa:tta Taxi에 오신것을 환영합니다!`);
+              history.push('/');
+            }
+            if (res.status === 401) {
+              alert('다시로그인 하세요');
+            }
+            if (res.status === 400) {
+              alert('이미 email로 가입된 기록이 있습니다.');
+            }
+          });
+      },
+      fail: function (err) {
+        alert('무언가 잘못되었습니다.');
+      },
+    });
+  };
+
+  const [isSignUp, setIsSignUp] = useState(false);
+
   return (
     <LoginContainer>
       <MovingIcon width="430" scrollamount="600">
         🚤
       </MovingIcon>
-      <Title>Welcome!</Title>
+      <Title>{isSignUp === true ? 'Join Wa:tta!' : 'Welcome!'}</Title>
       <SubText>한강을 누비자! 와따택시</SubText>
-      <KakaoLoginButton>
+      <KakaoLoginButton onClick={() => KakaoLogin()}>
```

Comment:
```suggestion
      <KakaoLoginButton onClick={KakaoLogin}>
```
굳이 즉시실행 함수로 만들어주지 않고 함수 자체를 넘겨주어도 같은 결과일 것 같습니다

Reply:
말씀해주신대로 바꾸면 처음 렌더 될때 한번 실행되는 것으로 알고 있고 실제로도 그렇게 되는데 어떤의미로 바꾸라고 해주신건지요?
질문드리러 가겠습니다.

Reply:
'즉시실행함수' 한번더 공부하고 수정후에 찾아뵙겠습니다!